Conch/lib/Conch/Controller/DeviceRoles.pm

```diff
@@ -0,0 +1,177 @@
+package Conch::Controller::DeviceRoles;
+
+use Mojo::Base 'Mojolicious::Controller', -signatures;
+
+use Conch::Models;
+
+
+=head2 get_all
+
+Get all device roles
+
+=cut
+
+sub get_all ($c) {
+	my $many = Conch::Model::DeviceRole->all();
+	$c->status(200, [ map { $_->TO_JSON } $many->@* ]);
+}
+
+
+=head2 get_one
+
+Get a single device service
+
+=cut
+
+sub get_one ($c) {
+	my $s = Conch::Model::DeviceRole->from_id($c->param('id'));
+	return $c->status(404 => { error => "Not found" }) unless $s;
+	return $c->status(404 => { error => "Not found" }) if $s->deactivated;
+
+	$c->status(200, $s->TO_JSON);
```

Comment:
Out of curiosity, is the call to `TO_JSON` necessary? Mojo::JSON should invoke it if present.